info-compiler/src/test/java/com/google/sps/InfoCompilerTest.java

```diff
@@ -0,0 +1,265 @@
+    datastore = datastoreHelper.getOptions().getService();
+    infoCompiler = new InfoCompiler(datastore);
+  }
+
+  @Test
+  public void queryCivicInformation_succeedWithMockResponse() throws Exception {
+    // Query the Civic Information API with a mock HTTP client + mock callback function of type
+    // {@code ResponseHandler<String>} that converts any {@code HttpResponse} response to {@code
+    // ELECTION_RESPONSE}, and subsequently convert {@code ELECTION_RESPONSE} to {@code json}.
+    // {@code httpGet} is irrelevant in this test, since {@code execute()} is mocked as above.
+    // Since String {@code ELECTION_RESPONSE} and JsonObject {@code ELECTION_JSON} match in
+    // content, {@code json} should be exactly the same as {@code ELECTION_JSON}. Here, we don't
+    // repeatedly test the same thing with {@code SINGLE_CONTEST_JSON}.
+    CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+    HttpGet httpGet = new HttpGet(ELECTION_QUERY_URL);
+    ArgumentCaptor<ResponseHandler<String>> argumentCaptor =
+        ArgumentCaptor.forClass(ResponseHandler.class);
+    when(httpClient.execute(anyObject(), argumentCaptor.capture())).thenReturn(ELECTION_RESPONSE);
+    JsonObject json =
+        infoCompiler.requestHttpAndBuildJsonResponseFromCivicInformation(httpClient, httpGet);
+    Assert.assertEquals(json, ELECTION_JSON);
+  }
+
+  @Test
+  public void storeBaseElectionInDatabase_checkDatastoreEntityConstructionFromJson()
+      throws IOException {
+    // Parse and re-structure base election information from {@code ELECTION_JSON}'s
+    // "elections" section and store the corresponding entity in the database. Said
+    // entity should contain information that is consistent with that in {@code ELECTION_JSON}.
+    // A Datastore emulator is used to simulate Datastore operations, as opposed to Mockito mocks.
+    JsonObject election = (JsonObject) ELECTION_JSON.getAsJsonArray("elections").get(0);
+    String[] yearMonthDay = election.get("electionDay").getAsString().split("-");
+    Date date = new Date(
+        Integer.parseInt(yearMonthDay[0]) - 1900,
+        Integer.parseInt(yearMonthDay[1]) - 1,
+        Integer.parseInt(yearMonthDay[2]),
+        4,
+        0);
+
+    infoCompiler.storeBaseElectionInDatabase(election);
+    Query<Entity> query =
+        Query.newEntityQueryBuilder()
+            .setKind("Election")
+            .build();
+    QueryResults<Entity> queryResult = datastore.run(query);
+    Assert.assertTrue(queryResult.hasNext());
+    Entity electionEntity = queryResult.next();
+    Assert.assertFalse(queryResult.hasNext());
+    Assert.assertEquals(electionEntity.getKey().getName(), election.get("name").getAsString());
+    Assert.assertEquals(electionEntity.getString("queryId"), election.get("id").getAsString());
+    Assert.assertEquals(electionEntity.getTimestamp("date").toDate(), date);
+    Assert.assertEquals(electionEntity.getList("candidatePositions"), Arrays.asList());
```

Comment:
Probably should have brought this up earlier, but Google's Truth java unit testing library is actually open source: https://truth.dev/benefits#readable-assertions

Allows you to do things like:
```
assertThat(electionEntity.getList("candidatePositions")).isEmpty();
```

Reply:
Up to you whether you want to go back and incorporate this...

Reply:
I can change it! Better readability is good